client/components/domains/use-my-domain/index.jsx

```diff
@@ -271,7 +271,13 @@ function UseMyDomain( props ) {
 				onConnect={
 					'auth_code' === domainAvailabilityData?.ownership_verification_type
 						? showOwnershipVerificationFlow
-						: onConnect
+						: () =>
+								onConnect &&
```

Comment:
Can we make a function called `handleConnect` that will do all of this stuff? Having it inline is really prone to error.

Reply:
Following @jdc91's suggestion to avoid prop drilling because `handleAddMapping` is only being called when the user uses their own domain, I just set the `signupDomainOrigin` directly from within `handleAddMapping`. This in turn means we don't have to do the inline logic anymore